src/lava/utils/float2fixed.py

```diff
@@ -1,26 +1,719 @@
+from lava.magma.core.process.ports.ports import OutPort
+
+import numpy as np
+import inspect
+
+
+class Float2FixedConverter:
+
+    def __init__(self):
+        self.floating_pt_rcfg = None
+        self.fixed_pt_rcfg = None
+        self.procs = None
+        self.hierarchical_procs = None
+        self.run_proc = None
+        self.var_ports = None
+        self.fixed_pt_proc_models = None
+        self.num_steps = None
+        self.quantiles = None
+        self.scale_domains = set()
+
+    def set_run_cfg(self, floating_pt_rcfg, fixed_pt_rcfg) -> None:
+        """Set run config for floating- and fixed-point ProcessModels
+        Parameters.
+        ----------
+
+        floating_pt_rcfg : RunConfig
+            RunConfig for floating-point ProcessModel
+        fixed_pt_rcfg : RunConfig
+            RunConfig for fixed-point ProcessModel
+        """
+
+        if not issubclass(type(floating_pt_rcfg), RunConfig):
+            raise TypeError("'floating_pt_rcfg' must be RunConfig")
+
+        if not issubclass(type(fixed_pt_rcfg), RunConfig):
+            raise TypeError("'fixed_pt_rcfg' must be RunConfig")
+
+        self.floating_pt_rcfg = floating_pt_rcfg
+        self.fixed_pt_rcfg = fixed_pt_rcfg
+
+    def convert(self, proc, num_steps=100, quantiles=[0, 1],
+                find_connected_procs=True):
+        """Convert Processes passed from floating-point to fixed-point model.
+        The floating- and fixed-point ProcessModels to be used must be
+        specificed via setting the run configurations.
+        After conversion has been executed, the converter parameter are stored
+        in the member 'scaled_params' of the class, a dictionary with keys
+        Process id and Variable name.
+
+        The conversion has the following steps:
+            1. Getting the Processes that need to be converted.
+            2. Fetching all Vars and OutPorts and RefPorts of the Processes.
+            3. If there is a hierarchical Process among the Processes to be
+               converted, instantiate new Processes copying the structure of
+               the passed model without hierarchical Processes. In the
+               following, the conversion will be performed on the new model.
+               Redo step 2 with the new model and continue with 4.
+            4. Finding and instantiating the fixed-point ProcessModels.
+            5. Retrieving the information needed for conversion from the
+               fixed-point ProcessModels chosen via the provided fixed-point
+               run configuration.
+            5. Creating monitors for recording the activation statistics of the
+            6  dynamical variables of Processes. With this, the domain of the
+               variable can be determined to constrain the floating- to
+               fixed-point conversion.
+            7. Execution of the floating-point model chosen via run
+               configuration. If non-default quantile information is passed the
```

Comment:
What's a non-default quantile information?

Reply:
Addressed in 58fd18a6c5ec776b81b991a6260f4edcc2d2ca2d.